atc/db/resource_config.go

```diff
@@ -281,19 +281,32 @@ func (r *ResourceConfigDescriptor) findWithParentID(tx Tx, parentColumnName stri
 	return id, true, nil
 }
 
-func findOrCreateResourceConfigScope(tx Tx, conn Conn, lockFactory lock.LockFactory, resourceConfig ResourceConfig, resource Resource, resourceTypes atc.VersionedResourceTypes) (ResourceConfigScope, error) {
+func findOrCreateResourceConfigScope(
+	tx Tx,
+	conn Conn,
+	lockFactory lock.LockFactory,
+	resourceConfig ResourceConfig,
+	resource Resource,
+	resourceType string,
+	resourceTypes atc.VersionedResourceTypes,
+) (ResourceConfigScope, error) {
+
 	var unique bool
 	var uniqueResource Resource
 	var resourceID *int
+
 	if resource != nil {
 		if !atc.EnableGlobalResources {
 			unique = true
 		} else {
-			customType, found := resourceTypes.Lookup(resource.Type())
+			customType, found := resourceTypes.Lookup(resourceType)
 			if found {
 				unique = customType.UniqueVersionHistory
 			} else {
-				unique = resourceConfig.CreatedByBaseResourceType().UniqueVersionHistory
+				baseType := resourceConfig.CreatedByBaseResourceType()
+				if baseType != nil {
+					unique = baseType.UniqueVersionHistory
```

Comment:
@pivotal-jwinters Is there a case where `baseType == nil`?

In that case `unique` will remain false, so the next condition will not be executed, and `resourceID` won't be set. I'm concerned that this could be potentially a null-dereference error.

Reply:
@wagdav `resourceID` being `nil` is OK - it will only be non-nil if the type has the "unique version history" flag, which strictly means that the scope is not associated to any particular resource ID.

Not sure about the semantics of when a `baseType` would be `nil` *and* a custom type not being found though.

Reply:
@vito  Thanks for the explanation!

Reply:
@wagdav @vito the case where baseType is nil and the custom type is not found **should never happen**™. It's a good point though, maybe it's actually better to crash in that case, because it might be masking something more difficult to debug.

Reply:
Yeah maybe let's have it return an error in that case? It seems super unlikely but I'd rather have an error to look for than silent strange behavior.

Reply:
yeah, i guess we can handle our errors properly instead of crashing lol

Reply:
fixed in f0bb0d2